### PR TITLE
Allow to define the page that must be shown when the queue is empty

### PR DESCRIPTION
**Pull request by framawiki**
Can we use something better than PageEmptyQueue != "" ?

Bug: https://phabricator.wikimedia.org/T166079

### CODE REVIEW

File: huggle/mainwindow.cpp [hunk 1]
```diff
@@ -1501,13 +1501,13 @@ void MainWindow::OnTimerTick0()
 void MainWindow::on_actionNext_triggered()
 {
     if (!this->Queue1->Next())
-        this->ShowCat();
```
Comment (benapetr) [inline huggle/mainwindow.cpp]: If you really need to rename it, I think ShowEmptyQueuePage would make more sense, this function sounds like it sets a literally empty page, which is confusing

File: huggle/mainwindow.cpp [hunk 2]
```diff
@@ -1908,10 +1908,18 @@ void MainWindow::DisplayNext(Query *q)
...
-    this->EnableEditing(false);
+    if(hcfg->ProjectConfig->PageEmptyQueue != "")
+    {
+        WikiPage *empty = new WikiPage(hcfg->ProjectConfig->PageEmptyQueue);
+        this->Browser->DisplayPreFormattedPage(empty);
+        this->LockPage();
+        delete empty;
+        this->Render();
+    }
+    else
```
Comment (benapetr) [inline huggle/mainwindow.cpp]: you forgot to add EnableEditing to false here, that is important

File: huggle/mainwindow.cpp [hunk 3]
```diff
@@ -1908,10 +1908,18 @@ void MainWindow::DisplayNext(Query *q)
     }
 }
 
-void MainWindow::ShowCat()
+void MainWindow::ShowEmptyPage()
 {
-    this->Browser->RenderHtml(Resources::Html_EmptyList);
-    this->EnableEditing(false);
+    if(hcfg->ProjectConfig->PageEmptyQueue != "")
```
Comment (benapetr) [inline huggle/mainwindow.cpp]: `!hcfg->ProjectConfig->PageEmptyQueue.isEmpty()`

Comment (benapetr) [inline huggle/mainwindow.cpp]: that function is faster

File: huggle/projectconfiguration.cpp [hunk 4]
```diff
@@ -174,6 +174,7 @@ bool ProjectConfiguration::Parse(QString config, QString *reason, WikiSite *site
     // Welcoming
     this->Welcome = HuggleParser::ConfigurationParse("welcome", config);
     this->WelcomeMP = HuggleParser::ConfigurationParse("startup-message-location", config, "Project:Huggle/Message");
+    this->PageEmptyQueue = HuggleParser::ConfigurationParse("emptyqueue-message-location", config, "");
```
Comment (benapetr) [inline huggle/projectconfiguration.cpp]: Project configuration? Why? This should go to user configuration, we don't want to override this setting per whole project, it should be optional for users who want to change that cat.

Also if you really want people to use this feature, information that this page can be customized should be inserted to default page

File: huggle/mainwindow.cpp (hunk 1 above)
Comment (framawiki) [inline huggle/mainwindow.cpp]: Ok.

File: huggle/mainwindow.cpp (hunk 2 above)
Comment (framawiki) [inline huggle/mainwindow.cpp]: Ok. I copied `DisplayWelcomeMessage()`, it may also need to be amended

**Comment by framawiki** [comment]
Thanks for your comments, here is a new version.

**Pull request merged**